### ISSUE TITLE
Running amd64 neo4j  container gets randomly stuck on M1

### ISSUE THREAD

**Issue by arjantop-cai** [docker/for-mac#6060]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

The container starts and is can be connected to (localhost:7474 in browser).

Expected output:
```
Selecting JVM - Version:11.0.13, Name:OpenJDK 64-Bit Server VM, Vendor:Oracle Corporation
Changed password for user 'neo4j'.
2021-11-24 16:03:34.458+0000 INFO  Starting...
2021-11-24 16:03:49.904+0000 INFO  ======== Neo4j 4.3.7 ========
2021-11-24 16:04:01.565+0000 INFO  Initializing system graph model for component 'security-users' with version -1 and status UNINITIALIZED
2021-11-24 16:04:01.773+0000 INFO  Setting up initial user from `auth.ini` file: neo4j
2021-11-24 16:04:01.778+0000 INFO  Creating new user 'neo4j' (passwordChangeRequired=false, suspended=false)
2021-11-24 16:04:01.829+0000 INFO  Setting version for 'security-users' to 3
2021-11-24 16:04:01.840+0000 INFO  After initialization of system graph model component 'security-users' have version 3 and status CURRENT
2021-11-24 16:04:01.862+0000 INFO  Performing postInitialization step for component 'security-users' with version 3 and status CURRENT
2021-11-24 16:04:02.866+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2021-11-24 16:04:11.070+0000 INFO  Remote interface available at http://localhost:7474/
2021-11-24 16:04:11.099+0000 INFO  Started.
```

### Actual behavior
The container startup gets "stuck" at a random point during startup, the container can still be exec-ed into but java process is stuck. Strace output is attached.

### Information
macOS Version: 12.0.1
Intel chip or Apple chip: Apple M1
Docker Desktop Version: 4.2.0

Docker desktop resources:
CPUS: 8
Memory: 32GB

### Steps to reproduce the behavior
Run the below command a couple of times:
```
docker run --rm -p7474:7474 -p7687:7687 --env=NEO4J_AUTH=neo4j/test --platform=linux/amd64 neo4j:latest
```
[neo4j-frozen.log](https://github.com/docker/for-mac/files/7597244/neo4j-frozen.log)

**Comment by Enzo90910** [comment]
May be related: I have the same behaviour with a amd64 Java container running hdfs (Docker Hub cybermaggedon/hadoop:2.10.0 built from fedora:32 and adding java-1.8.0-openjdk) on a 16GB M1 Pro laptop.

**Comment by arjantop-cai** [comment]
Can also confirm the same for Kafka

**Comment by Phlamethrower** [comment]
I'm also seeing this with a Java amd64 image on a M1 mac mini.

* Image: swaggerapi/swagger-codegen-cli:2.4.21 (which uses openjdk:8-jre-alpine as the base)
* Docker Desktop 4.3.0 (71786) / Engine 20.10.11 on macOS 12.0.1

Running the image with no args should print out some help text and then exit, but instead it just hangs, with activity monitor showing 100% CPU usage by qemu-system-aarch64.

I'm fairly certain the same image used to work fine with macOS 11 & an older version of Docker 4.x on the same mac, but I don't know the specific versions that worked.

**Comment by longwa** [comment]
This could also be related to the QEMU force multicore option, which produces unreliable emulation on the M1 and should be disabled until the memory model issues in QEmu can be sorted.

https://github.com/docker/for-mac/issues/6016

**Comment by voxmaster** [comment]
Similar problem on 16GB M1 Pro when building Java applications. Randomly stuck with qemu-system-aarch64 200% CPU load. After several restarts application builds successfully.

**Comment by ronildosouzafb** [comment]
Similar problem on 32GB M1 Pro when running Java applications. Randomly stuck.

**Comment by yogevyuval** [comment]
Similar issue with Logstash on M1, java process stuck during runtime / build

**Comment by lr-mike-chai** [comment]
Similar issue on 32GB M1 Pro for openapitools/openapi-generator-cli. Bashing into the container and running "java -version" or anything with "java" just stalls/freezes the terminal and nothing happens.

**Comment by pcallahan-r7** [comment]
I have seen apache tomcat containers using amd64 java images randomly freeze up on the m1.

**Comment by dgauche** [comment]
I am seeing the same on our own custom images making use of Java and Tomcat.

The issue is inconsistency. 1/5 times the container will boot and will run fine, but the other 4/5 times it will get stuck somewhere during the startup phase.

These images all run perfectly fine and reliably on intel based macs.

M1 Max 32gb

**Comment by ashi009** [comment]
I can confirm the same issue when running with Apple's Virtualization Framework.

**Comment by waterscar** [comment]
Same issue with docker for Mac version 4.19, and OpenJDK version "11.0.17" 2022-10-18 LTS

**Comment by equal-matt** [comment]
Another confirmation here!

**Comment by TheCycoONE** [comment]
The experimental Rosetta option seems to be a workaround. We've been recommending at my company with good results.

**Comment by keisukekomeda** [comment]
In my environment (M2 Pro, Docker Desktop 4.25.2 (129061), virtualization framework enabled), disabling Rosetta option fixes the problem.

**Comment by dgageot** [comment]
Hi everyone! Has anyone been able to reproduce with Docker Desktop 4.26? It contains a lot of fixes to the emulation layer, especially when Rosetta is enabled.

**Comment by dgageot** [comment]
Hey everyone. I'm going to close this issue because I believe it's been fixed with Docker Desktop 4.26. Feel free to ping me if you still see random crashes.
